frontend/app/src/App.test.tsx

```diff
@@ -931,24 +931,24 @@ describe("App.onHistoryChange", () => {
     pushStateSpy.mockRestore()
   })
 
-  it("does rerun when we are navigating to a different page and the last window history url contains an anchor", () => {
+  it("does rerun when we are navigating to a different page and the last window history url contains an anchor", async () => {
     const pushStateSpy = jest.spyOn(window.history, "pushState")
-
-    jest.spyOn(instance, "onPageChange")
+    const pageChangeSpy = jest.spyOn(instance, "onPageChange")
 
     // navigate to current page with anchor
     window.history.pushState({}, "", "#foo_bar")
     instance.onHistoryChange()
-    expect(instance.onPageChange).not.toHaveBeenCalled()
+    expect(pageChangeSpy).not.toHaveBeenCalled()
 
     // navigate to new page
     instance.handleNewSession(
       new NewSession({ ...NEW_SESSION_JSON, pageScriptHash: "sub_hash" })
     )
-    window.history.back()
+    instance.onHistoryChange()
```

Comment:
nit: I think calling `window.history.back()` here was intentional since it's a closer simulation of what happens in a real browser (`instance.onHistoryChange` should be called as a result of the user clicking the back button)